### PR TITLE
Port dogate features to nozzle

### PR DESCRIPTION
**Pull request by zippolyte**
*Note: Please remember to review the Datadog [Contribution Guidelines](https://github.com/DataDog/dd-agent/blob/master/CONTRIBUTING.md)
if you have not yet done so.*

### What does this PR do?

Port the features of [dogate](https://github.com/pivotal-cloudops/dogate) to the nozzle.
The only thing missing is the tag remapping: `job names | zookeeper-partition-8d3fab00d189bbaf7250 | zookeeper-z1`, since we do not have the availability zone index

### Motivation

What inspired you to submit this pull request?

### Testing Guidelines

An overview on [testing](https://github.com/DataDog/dd-agent/blob/master/tests/README.md)
is available in our contribution guidelines.

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: datadogclient/formatter.go [hunk 1]
```diff
@@ -33,6 +34,10 @@ func (f Formatter) Format(prefix string, maxPostBytes uint32, data map[metrics.M
 func (f Formatter) formatMetrics(prefix string, data map[metrics.MetricKey]metrics.MetricValue) []byte {
 	s := []metrics.Series{}
 	for key, mVal := range data {
+		// dogate feature
+		if strings.HasPrefix(key.Name, "bosh.healthmonitor") {
```
Comment (gmmeyer) [inline datadogclient/formatter.go]: I think you misread the feature:

```ruby
entry['metric'].sub!('datadog.nozzle.bosh-hm-forwarder', 'bosh.healthmonitor')
```

File: datadogclient/formatter_test.go [hunk 2]
```diff
@@ -33,4 +33,16 @@ var _ = Describe("Formatter", func() {
 
 		Expect(result).To(HaveLen(1))
 	})
+
+	It("does not prepend prefix to `bosh.healthmonitor`", func() {
```
Comment (gmmeyer) [inline datadogclient/formatter_test.go]: should we not do this or potentially should we send a dupe metric with the newer name?

File: integration_test/integration_test.go [hunk 3]
```diff
@@ -115,23 +118,28 @@ var _ = Describe("DatadogFirehoseNozzle", func() {
...
 				} else {
 					panic("Unknown tag")
 				}
 			} else if metric.Metric == "cloudfoundry.nozzle.origin.counterName" || metric.Metric == "cloudfoundry.nozzle.counterName" {
-				Expect(metric.Tags).To(HaveLen(4))
-				Expect(metric.Tags[0]).To(Equal("deployment:deployment-name"))
-				Expect(metric.Tags[1]).To(Equal("job:doppler"))
+				Expect(metric.Tags).To(HaveLen(7))
+				Expect(metric.Tags[0]).To(Equal("deployment:deployment-name-aaaaaaaaaaaaaaaaaaaa"))
+				Expect(metric.Tags[1]).To(Equal("deployment:deployment-name_env_name"))
```
Comment (gmmeyer) [inline integration_test/integration_test.go]: can we also add the tag `deployment:deployment-name`?

File: datadogclient/formatter.go (hunk 1 above)
Comment (zippolyte) [reply datadogclient/formatter.go]: Actually no, this is combined with https://github.com/DataDog/datadog-firehose-nozzle/pull/48/files#diff-cd6782160e51ec84de9b84e784c25799R47, for the full feature
one part replaces the metric name, and the other prevent appending the prefix `datadog.nozzle`

File: integration_test/integration_test.go (hunk 3 above)
Comment (zippolyte) [reply integration_test/integration_test.go]: done

File: datadogclient/formatter.go (hunk 1 above)
Comment (gmmeyer) [reply datadogclient/formatter.go]: ahh okay

File: integration_test/integration_test.go (hunk 3 above)
Comment (gmmeyer) [reply integration_test/integration_test.go]: thanks!

File: datadogclient/formatter.go (hunk 1 above)
Comment (gmmeyer) [reply datadogclient/formatter.go]: thanks!

**Pull request merged**